autogen/token_count_utils.py

```diff
@@ -106,34 +110,18 @@ def _num_token_from_messages(messages: Union[List, Dict], model="gpt-3.5-turbo-0
     except KeyError:
         logger.warning(f"Model {model} not found. Using cl100k_base encoding.")
         encoding = tiktoken.get_encoding("cl100k_base")
-    if model in {
-        "gpt-3.5-turbo-0613",
-        "gpt-3.5-turbo-16k-0613",
-        "gpt-4-0314",
-        "gpt-4-32k-0314",
-        "gpt-4-0613",
-        "gpt-4-32k-0613",
-    }:
-        tokens_per_message = 3
-        tokens_per_name = 1
-    elif model == "gpt-3.5-turbo-0301":
+    if "gpt-3" in model or "gpt-4" in model or model.startswith("o1"):
         tokens_per_message = 4  # every message follows <|start|>{role/name}\n{content}<|end|>\n
-        tokens_per_name = -1  # if there's a name, the role is omitted
-    elif "gpt-3.5-turbo" in model:
-        logger.info("gpt-3.5-turbo may update over time. Returning num tokens assuming gpt-3.5-turbo-0613.")
-        return _num_token_from_messages(messages, model="gpt-3.5-turbo-0613")
-    elif "gpt-4" in model:
-        logger.info("gpt-4 may update over time. Returning num tokens assuming gpt-4-0613.")
-        return _num_token_from_messages(messages, model="gpt-4-0613")
+        tokens_per_name = 1  # OpenAI guidance is 1 extra token if 'name' field is used
     elif "gemini" in model:
-        logger.info("Gemini is not supported in tiktoken. Returning num tokens assuming gpt-4-0613.")
-        return _num_token_from_messages(messages, model="gpt-4-0613")
+        logger.info("Gemini is not supported in tiktoken. Returning num tokens assuming gpt-4.")
```

Comment:
Is this change necessary? gpt-4-0613 is more specific than gpt-4. Or is the model changed?

Reply:
I'm not sure if makes a difference as all gpt-4 models behave the same way. Would you prefer I change it back to the specific one?

Reply:
Yes, I think we should keep the original, so that people know our counting is relying on an older version. May add the year "assuming gpt-4-0613 (2023)"

Reply:
Thanks, no problem, I'll update to gpt-4-0613 and use the noted text.

Reply:
Okay, I've committed that change, thanks